ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -242,7 +242,12 @@ densityDisconnect (GenesisWindow sgen) (SecurityParam k) states candidateSuffixe
           -- that number of remaining slots.
           potentialSlots =
             if idling || hasBlockAfter then 0
-            else sgen - totalBlockCount
+            else unknownTrailingSlots
+
+          -- Number of trailing slots in the genesis window that could have
+          -- headers which haven't been sent yet
+          unknownTrailingSlots = unSlotNo $
+            firstSlotAfterGenesisWindow - succWithOrigin (AF.headSlot fragment)
```

Comment:
Maybe deserves a comment that this shall only be used when `not hasBlockAfter` is satisfied, as it will otherwise underflow.

Could use [`countSlots`](https://ouroboros-consensus.cardano.intersectmbo.org/haddocks/ouroboros-consensus/Ouroboros-Consensus-HardFork-History-Util.html#v:countSlots) here, also has an assertion that this doesn't underflow.

Reply:
Since `fragment` is clipped to the genesis window, it is not possible for this to underflow.

Reply:
```suggestion
          unknownTrailingSlots = unSlotNo $
            -- Cannot underflow as @fragment@ is clipped to the genesis window
            firstSlotAfterGenesisWindow - succWithOrigin (AF.headSlot fragment)
```

Reply:
Right, thanks, this is using `fragment`, not `candidateSuffix` (the latter is *not* clipped). I think these could be renamed for more clarity.

Reply:
Renamed `fragment` to `clippedFragment` in 04841a9.